src/test/no_pmix_launcher_client.c

```diff
@@ -46,12 +46,101 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <semaphore.h>
-#include <gurt/common.h>
 #include <cart/api.h>
 
+#include "crt_internal.h"
 #include "no_pmix_launcher_common.h"
 #include "tests_common.h"
 
+struct wfr_status {
+	sem_t	sem;
+	int	rc;
+};
+
+static void
+rpc_handle_reply(const struct crt_cb_info *info)
+{
+        sem_t   *sem;
+
+        D_ASSERTF(info->cci_rc == 0, "rpc response failed. rc: %d\n",
+                info->cci_rc);
+
+        sem = (sem_t *)info->cci_arg;
+        sem_post(sem);
+}
+
+static void
+wfr_handle_reply(const struct crt_cb_info *info)
+{
+	struct wfr_status	*wfrs;
+
+	wfrs = (struct wfr_status *)info->cci_arg;
+
+	wfrs->rc = info->cci_rc;
+
+        sem_post(&wfrs->sem);
+}
+
+static int wait_for_ranks(crt_context_t ctx, crt_group_t *grp, d_rank_list_t *rank_list, int tag, double timeout)
+{
+	struct wfr_status	ws;
+	d_rank_t		rank;
+	crt_rpc_t		*rpc = NULL;
+	crt_endpoint_t          server_ep;
+	int                     i = 0;
+	int                     rc = 0;
+
+        rc = sem_init(&ws.sem, 0, 0);
+        D_ASSERTF( rc == 0, "sem_init() failed; rc=%d\n", rc);
```

Comment:
(style)  code indent should use tabs where possible
(style)  space prohibited after that open parenthesis '('